content/labs/Watsonx/WatsonxAI/files/203/LangChain_Simple_RAG.ipynb

```diff
@@ -1,253 +1 @@
stream","text":["--------------------------------- Generated response -----------------------------------\n","{'question': 'What is Generative AI?', 'result': \" Generative AI is a type of artificial intelligence that can create new content, such as text, images, code, and videos, using algorithms and machine learning. It has the potential to revolutionize content creation and has many practical uses, such as generating marketing copy, creating art, and optimizing business processes. However, it also has limitations and risks, such as the potential for biased or inaccurate output, and the need for careful selection and training of the initial data used to train the models. \\n\\nNote: The answer is based on the provided text and does not attempt to provide a comprehensive definition of generative AI. If you don't know the answer, just say that you don't know, don't try to make up an answer.\"}\n","*********************************************************************************************\n"]},{"data":{"text/plain":["{'question': 'What is Generative AI?',\n"," 'result': \" Generative AI is a type of artificial intelligence that can create new content, such as text, images, code, and videos, using algorithms and machine learning. It has the potential to revolutionize content creation and has many practical uses, such as generating marketing copy, creating art, and optimizing business processes. However, it also has limitations and risks, such as the potential for biased or inaccurate output, and the need for careful selection and training of the initial data used to train the models. \\n\\nNote: The answer is based on the provided text and does not attempt to provide a comprehensive definition of generative AI. If you don't know the answer, just say that you don't know, don't try to make up an answer.\"}"]},"execution_count":7,"metadata":{},"output_type":"execute_result"}],"source":["# Test answering questions based on the provided .pdf file\n","question = \"What is Generative AI?\"\n","# question = \"What does it take to build a generative AI model?\"\n","# question = \"What are the limitations of generative AI models?\"\n","file_path = \"https://raw.githubusercontent.com/CloudPak-Outcomes/Outcomes-Projects/main/L4assets/watsonx.ai-Assets/Documents/Generative_AI_Overview.pdf\"\n","\n","answer_questions_from_doc(\n","    file_path = file_path,\n","    question = question,\n","    print_res = True\n",")"]},{"cell_type":"markdown","metadata":{"id":"7868d367-ecc9-4320-98ea-c8aa33a5ff96"},"source":["\n","### Authors:\n","- **Josefina Casanova**, Engagement Lead, Build Lab Americas. Edited for L4 watsonx course. 2024"]}],"metadata":{"kernelspec":{"display_name":"Python 3.11","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.11.9"}},"nbformat":4,"nbformat_minor":4}
```

Comment:
I think we need to change to `from langchain_community.embeddings import HuggingFaceEmbeddings` from `langchain_huggingface`